File: megumi/src/main/java/com/erisu/cloud/megumi/event/GroupSchedulerEvent.java
package com.erisu.cloud.megumi.event;

import lombok.extern.slf4j.Slf4j;
import net.mamoe.mirai.event.*;
import net.mamoe.mirai.event.events.BotActiveEvent;
import net.mamoe.mirai.event.events.BotOnlineEvent;
import net.mamoe.mirai.message.data.PlainText;

/**
 * @Description 定时任务事件插件
 * @Author alice
 * @Date 2020/12/30 11:11
 **/
@Event
@Slf4j
public class GroupSchedulerEvent extends SimpleListenerHost {


    /**
     * 什么bug，为什么进不来login能进relogin，不看了
     *
     * @param event
     * @return
     */

    @EventHandler
    public ListeningStatus onBotOnline(BotOnlineEvent event) {
        event.getBot().getGroup(604515343).sendMessage(new PlainText("我来了我来了我来了"));
        return ListeningStatus.STOPPED;
    }


    @EventHandler(priority = EventPriority.NORMAL)
    public ListeningStatus onBotActive(BotActiveEvent event) {
//        event.getBot().getGroup(604515343).sendMessage(new PlainText(""));
        return ListeningStatus.LISTENING;
    }


}

File: megumi/src/main/java/com/erisu/cloud/megumi/event/GroupSchedulerEvent.kt
package com.erisu.cloud.megumi.event

import lombok.extern.slf4j.Slf4j
import net.mamoe.mirai.event.EventHandler
import net.mamoe.mirai.event.ListeningStatus
import net.mamoe.mirai.event.SimpleListenerHost
import net.mamoe.mirai.event.events.BotOnlineEvent
import net.mamoe.mirai.message.data.PlainText

/**
 *@Description 定时任务事件插件
 *@Author alice
 *@Date 2022/5/10 15:52
 **/
@Event
@Slf4j
class GroupSchedulerEvent : SimpleListenerHost() {
    //    /**
//     *
//     *
//     * @param event
//     * @return
//     */
    @EventHandler
    suspend fun onBotOnline(event: BotOnlineEvent): ListeningStatus {
        event.bot.getGroup(604515343)!!.sendMessage(PlainText("我来了我来了我来了"))
        return ListeningStatus.LISTENING
    }


//    @EventHandler(priority = EventPriority.NORMAL)
//    fun onBotActive(event: BotActiveEvent?): ListeningStatus? {
////        event.getBot().getGroup(604515343).sendMessage(new PlainText(""));
//        return ListeningStatus.LISTENING
//    }
}